cmd/readiness/testdata/health-status-deadlocked-with-prev-config.json

```diff
@@ -0,0 +1,159 @@
+              "moveDoc": "Wait for the replica set to be initialized by another member",
+              "steps": [
+                {
+                  "step": "WaitRsInit",
+                  "stepDoc": "Wait for the replica set to be initialized by another member",
+                  "isWaitStep": true,
+                  "started": "2023-01-19T17:27:21.672605664Z",
+                  "completed": null,
+                  "result": "error"
+                }
+              ]
+            },
+            {
+              "move": "WaitFeatureCompatibilityVersionCorrect",
+              "moveDoc": "Wait for featureCompatibilityVersion to be right",
+              "steps": [
+                {
+                  "step": "WaitFeatureCompatibilityVersionCorrect",
+                  "stepDoc": "Wait for featureCompatibilityVersion to be right",
+                  "isWaitStep": true,
+                  "started": null,
+                  "completed": null,
+                  "result": ""
+                }
+              ]
+            }
+          ]
+        },
+        {
+          "automationConfigVersion": 2,
+          "started": "2023-01-19T17:36:34.742889301Z",
+          "completed": "2023-01-19T17:36:47.913043483Z",
+          "moves": [
+            {
+              "move": "WaitHasCorrectAutomationCredentials",
+              "moveDoc": "Wait for the automation user to be added (if needed)",
+              "steps": [
+                {
+                  "step": "WaitHasCorrectAutomationCredentials",
+                  "stepDoc": "Wait for the automation user to be added (if needed)",
+                  "isWaitStep": true,
+                  "started": "2023-01-19T17:36:34.742906201Z",
+                  "completed": null,
+                  "result": "wait"
+                }
+              ]
+            }
+          ]
+        },
+        {
+          "automationConfigVersion": 3,
+          "started": "2023-01-19T17:38:33.622622261Z",
+          "completed": null,
+          "moves": [
+            {
+              "move": "ChangeVersion",
+              "moveDoc": "Change MongoDB Version",
+              "steps": [
+                {
+                  "step": "CheckWrongVersion",
+                  "stepDoc": "Check that MongoDB version is wrong",
+                  "isWaitStep": false,
+                  "started": "2023-01-19T17:38:33.622638561Z",
+                  "completed": "2023-01-19T17:38:33.622959367Z",
+                  "result": "success"
+                },
+                {
+                  "step": "CheckRsCorrect",
+                  "stepDoc": "Check that replica set configuration is correct",
+                  "isWaitStep": false,
+                  "started": "2023-01-19T17:38:33.622960067Z",
+                  "completed": "2023-01-19T17:38:33.623363973Z",
+                  "result": "success"
+                },
+                {
+                  "step": "WaitCanUpdate",
```

Comment:
Our test fail here. Agents receive a new automation config with above step. By current definition they are never in `inGoalState` since they are waiting for the update and therefore `unready`. Depending on how fast the agent is able to realise updates the pod can get unhealthy and therefore the cluster. Fully stopping the update due to sts controller not continuing

Reply:
I verified the following:
- all agents receive the new automation config containing the `WaitCanUpdate`
- agents are getting replaced one after other and therefore `lose` the `WaitCanUpdate` step, since they were replaced
- the primary got replaced last, giving it more time to get unhealthy